src/zenml/models/v2/core/run_metadata.py

```diff
@@ -71,23 +64,53 @@ class Config:
 class RunMetadataResponseBody(WorkspaceScopedResponseBody):
     """Response body for run metadata."""
 
-    key: str = Field(
-        title="The key of the metadata.",
-        max_length=STR_FIELD_MAX_LENGTH,
-    )
+    key: str = Field(title="The key of the metadata.")
     value: MetadataType = Field(
-        title="The value of the metadata.",
-        max_length=TEXT_FIELD_MAX_LENGTH,
-    )
-    type: MetadataTypeEnum = Field(
-        title="The type of the metadata.",
-        max_length=STR_FIELD_MAX_LENGTH,
+        title="The value of the metadata.", union_mode="smart"
     )
+    type: MetadataTypeEnum = Field(title="The type of the metadata.")
```

Comment:
```suggestion
    type: MetadataTypeEnum = Field(title="The type of the metadata.", union_mode="smart")
```
Maybe?

Reply:
Since the `MetadataTypeEnum` is not a `Union`, this would fail with: `RuntimeError: Unable to apply constraint union_mode to schema enum`.

But with a similar mindset, I already applied `union_mode="smart"` to all the `MetadataType` fields.

Reply:
Too many files, confused it with `MetadataType`